airflow/providers/databricks/hooks/databricks.py

```diff
@@ -404,6 +479,40 @@ def start_cluster(self, json: dict) -> None:
         """
         self._do_api_call(START_CLUSTER_ENDPOINT, json)
 
+    def activate_cluster(self, json: dict, polling: int, timeout: int | None = None) -> None:
+        """
+        Start the cluster, and wait for it to be ready.
+
+        :param json: json dictionary containing cluster specification.
+        :param polling: polling interval in seconds.
+        :param timeout: timeout in seconds. -1 means no timeout.
+        """
+        cluster_id = json['cluster_id']
+
+        api_called = False
+        elapsed_time = 0
+
+        while True:
+            run_state = self.get_cluster_state(cluster_id)
+
+            if run_state.is_running:
+                return
+            elif run_state.is_terminal:
+                if not api_called:
+                    self.start_cluster(json)
+                    api_called = True
+                else:
+                    raise AirflowException(
+                        f"Cluster {cluster_id} start failed with '{run_state.state}' state: {run_state.state_message}"
+                    )
+
+            # wait for cluster to start
+            time.sleep(polling)
+
+            elapsed_time += polling
+            if timeout and elapsed_time <= timeout:
+                raise AirflowException(f"Cluster {cluster_id} start timed out after {timeout} seconds")
```

Comment:
you assume that the time in each iteration is `polling`, but there is the time needed to call the API and get the state; you can replace this timeout by `max_attempts` or implement a real timeout based on a start_date defined before entering the loop.

Reply:
Yes! you're right! I will change my code to reflect "real" timeout! Thanks! 👍

Reply:
I've changed it :)
https://github.com/apache/airflow/pull/34071/commits/0745545a160f54fbe6f8e42b94a839fe0655945a